### ISSUE TITLE
too much cpu use

### ISSUE THREAD

**Issue by zhangbufan** [hsiafan/httpdump#39]
too much cpu use

**Comment by kelvinkuo** [comment]
yes, I meet the same problem and I pprof the program, its caused by discardAll func
![20200505143508](https://user-images.githubusercontent.com/2784354/81040292-a7104780-8edd-11ea-9049-e751f0452325.jpg)
